build-system/babel-plugins/babel-plugin-transform-log-methods/index.js

```diff
@@ -0,0 +1,288 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Indirects log messages through expansion calls with
+ * base36-encoded message ids. These redirect to a URL or look up a message
+ * table to interpolate and display the logged string.
+ *
+ *    dev().assert(foo != bar, 'foo should not be bar')
+ *    // transforms into 👇
+ *    dev().assert(foo != bar, ['1z'])
+ *
+ * `assert is overloaded so array syntax causes lookup of the message template
+ * and expansion of the message, or displaying a URL where the interpolated
+ * message is displayed.
+ *
+ * Additionally outputs a JSON file containing [template, id] pairs keyed by
+ * message template. This table is "backwards" for deduping, to key by id its
+ * rows should be reversed separately.
+ *
+ * When the template has arguments they are nested into the expansion method:
+ *
+ *    dev().assert(false, '%s should not be %s.', foo, bar)
+ *    // transforms into 👇
+ *    dev().assert(false, ['e2', foo, bar])
```

Comment:
We might be able to save more bytes by passing these as position arguments, instead of in an array:

```js
// old
dev().assert(false, ['e2', foo, bar])

// new
dev().assert(false, 'e2', foo, bar)